Documentation/contributors/design-proposals/operator-config.md

```diff
@@ -0,0 +1,338 @@
+    - mountPath: /config
+      name: config-volume
+    # provide application config via env variables
+    env:
+    - name: SPECIAL_LEVEL_KEY
+      valueFrom:
+        configMapKeyRef:
+          name: special-config
+          key: special.how
+    envFrom:
+    - configMapRef:
+        name: etcd-env-config
+```
+
+### Subscription Status Changes
+
+The subscription status should reflect whether or not the configuration was successfully applied to the operator pods.
+
+New status conditions (abnormal-true):
+
+```yaml
+  conditions:
+  - message: "SPECIAL_LEVEL_KEY" couldn't be applied...
+    reason: EnvFailure
+    status: True
+    type: ConfigFailure
+  - message: No operator pods found matching selector.
+    reason: NoMatchingPods
+    status: True
+    type: PodConfigSelectorFailure
+```
+
+Reasons for `ConfigFailure`
+
+ - `EnvFailure`
+ - `EnvFromFailure`
+ - `VolumeFailure`
+ - `VolumeMountFailure`
+ - `TolerationFailure`
+ - `NodeSelectorFailure`
+ - `ResourceRequestFailure`
+ - `ResourceLimitFailure`
+
+### Implementation
+
+#### Subscription Spec and Status
+
+Spec and Status need to be updated to include the fields described above, and the openapi validation should be updated as well.
+
+#### Control Loops
+
+#### Install Strategy
+
+Most of the change will take place in the install strategy; which knows how to take the deployment spec defined in a CSV and check if the cluster is up-to-date, and apply changes if needed.
+
+- The install strategy will now need to accept the additional configuration from the subscription.
+	- `CheckInstalled` will need to combine the additional config with the deployment spec from the ClusterServiceVersion.
+		- Subscription config will overwrite the settings on the CSV.
+	- `Install` will also need to combien the additional config with the deployment spec.
+
+Appropriate errors should be returned so that we can construct the status that needed for the subscription config status conditions.
+
+This requires that https://github.com/operator-framework/operator-lifecycle-manager/pull/931 have merged, so that changes to a deployment are persisted to the cluster.
+	
+#### Subscription sync
+
+- Subscription sync will now need to use the install strategy for the installed CSV to determine if the config settings have been properly applied.
+	- If not, appropriate status should be written signalling what is missing, and the associated CSV should be requeued. 
+	- Requeing the CSV will attempt to reconcile the config with the deployment on the cluster.
+
+#### Openshift-specific implementation
+
+On start, OLM (catalog operator) needs to:
+
+- Check if the [openshift proxy config api](https://github.com/openshift/api/blob/master/config/v1/types_proxy.go) is available
+- If so, set up watches / informers for the GVK to keep our view of the global proxy config up to date.
+
+When reconciling an operator's `Deployment`:
+
+ - If the global proxy object is not set / api doesn't exist, do nothing different.
```

Comment:
Question: `If the global proxy object is not set` - technically there can be N (N>1) CR instance(s) of `Proxy` type in the cluster. Will the global proxy object have a predefined name or will there be only one instance of `Proxy`? @ecordell @shawn-hurley

Reply:
There will be many proxy objects, but the one named `cluster` should be considered the global proxy config.